homeassistant/components/vasttrafik/config_flow.py

```diff
@@ -0,0 +1,459 @@
+    ) -> ConfigFlowResult:
+        """Handle the initial step of adding a departure board."""
+        if user_input is not None:
+            search_query = user_input.get("search_query", "").strip()
+            if not search_query:
+                errors = {"base": "search_required"}
+            elif len(search_query) < 2:
+                errors = {"base": "search_too_short"}
+            else:
+                parent_entry = self._get_entry()
+                if not parent_entry:
+                    return self.async_abort(reason="parent_entry_not_found")
+
+                stations, error_code = await search_stations(
+                    self.hass, parent_entry, search_query
+                )
+                if error_code:
+                    errors = {"base": error_code}
+                elif not stations:
+                    errors = {"base": "no_stations_found"}
+                else:
+                    self._search_results = stations
+                    return await self.async_step_select_station()
+        else:
+            errors = {}
+
+        search_schema = vol.Schema(
+            {
+                vol.Required("search_query"): vol.All(str, vol.Length(min=2)),
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=search_schema,
+            errors=errors,
+        )
+
+    async def async_step_select_station(self, user_input=None):
+        """Select station and configure departure sensor."""
+        errors = {}
+
+        if user_input is not None:
+            selected_station = user_input.get("station")
+            if not selected_station:
+                errors["base"] = "station_required"
+            else:
+                self._selected_station = selected_station
+                return await self.async_step_configure()
+
+        station_options = {
+            station["value"]: station["label"] for station in self._search_results
+        }
+
+        station_schema = vol.Schema(
+            {
+                vol.Required("station"): vol.In(station_options),
+            }
+        )
+
+        return self.async_show_form(
+            step_id="select_station",
+            data_schema=station_schema,
+            errors=errors,
+        )
+
+    async def async_step_configure(self, user_input=None):
+        """Configure the departure sensor details."""
+        errors = {}
+
+        if user_input is not None:
+            lines = []
+            if user_input.get(CONF_LINES):
+                lines = [
+                    line.strip()
+                    for line in user_input[CONF_LINES].split(",")
+                    if line.strip()
+                ]
+
+            tracks = []
+            if user_input.get(CONF_TRACKS):
+                tracks = [
+                    track.strip()
+                    for track in user_input[CONF_TRACKS].split(",")
+                    if track.strip()
+                ]
```

Comment:
Instead of asking the user to input comma-separated lists of tracks and lines, can we use the textselector in multiple mode?

Reply:
Yeah, I was thinking of doing that but eventually it was so much other things I never got around to it. I will look into it.